torch/distributed/distributed_c10d.py

```diff
@@ -1393,6 +1396,7 @@ def _new_process_group_helper(
     _world.pg_map[pg] = (backend, prefix_store)
     _world.pg_names[pg] = group_name
     pg._set_group_name(group_name)
```

Comment:
can be done in follow up PRs, but I think maybe we can combine `pg._set_group_name` and `register_process_group` together, as I don't quite remember the usage of the former API.